### PR TITLE
Adding a pure numpy impl of PRNG split/fold so we don't require jax installs.

### PR DESCRIPTION
**Pull request by znado**
Also making a small fix for the Jax MNIST submission where `try_gcs=True` was slowing down our tfds reading.

Tested that can run the PyTorch MNIST example without `jax` installed, and that the Jax MNIST example still runs.

The general strategy is to import a seed manipulation module like:
```
import random_utils as prng
try:
  import jax.random as prng
except (ImportError, ModuleNotFoundError):
  logging.warning(
      'Could not import jax.random for the submission runner, falling back to '
      'numpy random_utils.')
```

And then call `prng.split`, `prng.fold_in`, `prng.PRNGKey`.

I'm open to suggestions if people prefer a different strategy for switching which implementation to use (`FLAGS`, etc.)!

### CODE REVIEW

**Comment by hanlint** [comment]
Thanks @znado , I am able to be unblocked by this PR. However, if we must maintain separate implementations, I would recommend switching based on a flag. With the current design, I need to explicitly `pip uninstall jax` to access the numpy rng. Otherwise, the pytorch code will silently use the jax rng and hit CUDA OOM.

Now that we have a functionally equivalent numpy rng though, is it still necessary to maintain a jax-based rng for the managing the seeds?

**Comment by znado** [comment]
> I would recommend switching based on a flag

Added `--use_jax_rng`.

> Now that we have a functionally equivalent numpy rng though, is it still necessary to maintain a jax-based rng for the managing the seeds?

The Flax models assume jax.PRNGKeys for their rng manipulation, which is unfortunate, but given the current setup I don't think it should be much maintenance work to keep the jax rngs around.

File: submission_runner.py [hunk 1]
```diff
@@ -66,8 +60,11 @@
 flags.DEFINE_string(
     'data_dir',
     '~/',
-    'Dataset location'
-)
+    'Dataset location')
+flags.DEFINE_boolean(
+    'use_jax_rng',
+    False,
+    'Whether to use the Jax or Numpy RNG library.')
```
Comment (hanlint) [inline submission_runner.py:67]: Can you just add a 'For Pytorch, use the Numpy RNG library' to the description here to motivate the differences for users?

Comment (znado) [reply submission_runner.py]: done!
